src/machine/machine_k210.go

```diff
@@ -0,0 +1,642 @@
+// +build k210
+
+package machine
+
+import (
+	"device/kendryte"
+	"errors"
+	"runtime/interrupt"
+)
+
+func CPUFrequency() uint32 {
+	return 390000000
+}
+
+type PinMode uint8
+type fpioaPullMode uint8
+type PinChange uint8
+
+// Pin modes.
+const (
+	PinInput PinMode = iota
+	PinInputPullUp
+	PinInputPullDown
+	PinOutput
+)
+
+// FPIOA internal pull resistors.
+const (
+	fpioaPullNone fpioaPullMode = iota
+	fpioaPullDown
+	fpioaPullUp
+)
+
+// GPIOHS pin interrupt events.
+const (
+	PinRising PinChange = iota + 1
+	PinFalling
+	PinToggle
+	PinHigh
+	PinLow = 8
```

Comment:
This wasn't very clear to me, especially with the regular enum that was used as a bitfield in `SetInterrupt`. Maybe this is more readable:

```suggestion
	PinRising PinChange = 1 << iota
	PinFalling
	PinHigh
	PinLow
	PinToggle = PinRising | PinFalling
```

Reply:
done